src/compiler/binder.ts

```diff
@@ -2656,7 +2666,7 @@ namespace ts {
 
             // If a FunctionDeclaration is async, then it is TypeScript syntax.
             if (modifierFlags & ModifierFlags.Async) {
-                transformFlags |= TransformFlags.AssertTypeScript;
+                transformFlags |= TransformFlags.AssertTypeScript | TransformFlags.AssertES2017;

```

Comment:
is this comment still accurate? Or are async functions valid ES2017 too? If so, they probably shouldn't be marked as typescript.